packages/itwinui-css/src/searchbox/searchbox.scss

```diff
@@ -1,8 +1,12 @@
 // Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 // See LICENSE.md in the project root for license terms and full copyright notice.
 
+@use '../utils/icon' as *;
+@use 'sass:map';
+
 .iui-expandable-searchbox {
-  inline-size: var(--_iui-input-min-height);
+  // button icon size + 2 * button padding + 2 * border width
+  inline-size: calc(#{map.get($icon-sizes, m)} + 2 * var(--iui-size-m) + 2 * 1px);
```

Comment:
> I'm concerned about the brittle logic for handling sizing here.
> 
> > Having a borderless button was necessary to make the animation between the collapsed and expanded state look smooth by using only the flex container border.

Yes, the logic is brittle, open to suggestions.

> Can you show how it looks using default button instead of borderless?

Here is the result with a default button (https://github.com/iTwin/iTwinUI/pull/1888/commits/093ca0d931ad1b45ef2832ebb4b3df23ca70f4bc):

https://github.com/iTwin/iTwinUI/assets/45748283/55e6d3e3-3b3a-4ea4-b232-b8d78746c657

The issue here is that the right border of the button is visible when going from expanded to collapsed.

Reply:
Here's my suggestion: Use the default button (not borderless) and remove the transition.

![image](https://github.com/iTwin/iTwinUI/assets/9084735/3d5ae726-5daa-4830-b012-cccbf9959204)

`transition: width` is already pretty bad for performance so we should not be using it in the first place. In a future PR, we can think about adding a more optimized transition (likely using JS).

Reply:
I noticed some other redundant code too.

<details><summary>Complete diff</summary>

```diff
--- a/packages/itwinui-css/src/searchbox/searchbox.scss
+++ b/packages/itwinui-css/src/searchbox/searchbox.scss
@@ -5,37 +5,14 @@
 @use 'sass:map';
 
 .iui-expandable-searchbox {
-  // button icon size + 2 * button padding + 2 * border width
-  inline-size: calc(#{map.get($icon-sizes, m)} + 2 * var(--iui-size-m) + 2 * 1px);
-  transition: width var(--iui-duration-1) ease-in-out;
-
-  &:has(.iui-searchbox-open-button:focus-visible) {
-    outline: 2px solid var(--_iui-input-accent-border-color);
-    outline-offset: -2px;
-  }
-
-  @supports not selector(:has(+ *)) {
-    &:where(.iui-searchbox-open-button:focus-visible) {
-      outline: 2px solid var(--_iui-input-accent-border-color);
-      outline-offset: -2px;
-    }
-  }
+  inline-size: auto;
+  padding-inline: 0; // To remove 1px padding from input-flex-container
 
   // Expanded
   &:where([data-iui-expanded='true']) {
     inline-size: 100%;
     border-color: var(--_iui-input-border-color);
-    padding-inline-start: 0; // To prevent layout shift when toggling between collapsed and expanded states
   }
-
-  // Collapsed
-  &:where(:not([data-iui-expanded='true'])) {
-    padding-inline: 0; // To center iui-searchbox-open-button inline in collapsed state
-  }
-}
-
-.iui-searchbox-open-button {
-  padding-inline: var(--iui-size-m); // To prevent the borderless button from becoming a square
 }
 
 // noop https://github.com/iTwin/iTwinUI/pull/1881#discussion_r1506166537

--- a/packages/itwinui-react/src/core/SearchBox/SearchBox.tsx
+++ b/packages/itwinui-react/src/core/SearchBox/SearchBox.tsx
@@ -312,7 +312,7 @@ SearchBoxCollapseButton.displayName = 'SearchBox.CollapseButton';
 // ----------------------------------------------------------------------------
 
 const SearchBoxExpandButton = React.forwardRef((props, ref) => {
-  const { children, className, onClick: onClickProp, ...rest } = props;
+  const { children, onClick: onClickProp, ...rest } = props;
 
   const {
     onExpand,
@@ -324,12 +324,11 @@ const SearchBoxExpandButton = React.forwardRef((props, ref) => {
   return (
     <SearchBoxButton
       ref={useMergedRefs(ref, openButtonRef)}
-      className={cx('iui-searchbox-open-button', className)}
       aria-label='Expand searchbox'
       size={sizeContext}
       disabled={isDisabled}
       onClick={mergeEventHandlers(onClickProp, onExpand)}
-      styleType='borderless'
+      styleType='default'
       {...rest}
     >
       {children ?? <SvgSearch />}

```

</details>

Reply:
Sounds good, applied the diff (https://github.com/iTwin/iTwinUI/pull/1888/commits/60d701b7e1a5fb22dbacddedc043a8d0478e5142). The result of this is:

https://github.com/iTwin/iTwinUI/assets/45748283/fd914f7a-75e5-4cf0-831e-ee4a16001a76

However, when wrapping it with a flex, since we now are no longer manually setting `inline-size` to some exact value, when wrapped with a `display: flex`, it could give some unexpected results like showing the container even when the SearchBox is not expanded:

<details>
<summary>Code</summary>

```jsx
 <div
  style={{
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'flex-start',
  }}
>
  <SearchBox expandable></SearchBox>
  <p>foobar</p>
</div>
```

</details>

Without `alignItems: 'flex-start'`:

https://github.com/iTwin/iTwinUI/assets/45748283/d17e1db3-0c7b-4a33-9fac-a31d336bcd32

With `alignItems: 'flex-start'`:

https://github.com/iTwin/iTwinUI/assets/45748283/7b6734dc-2f7b-4f52-91f2-9c81b6b6d3eb

Reply:
I think you can use `inline-size: fit-content` instead of `auto`?

Reply:
Yes, changed to `fit-content` (https://github.com/iTwin/iTwinUI/pull/1888/commits/9e2d128ba61ad2ad10dcd9bb73f47d1e1b4fb121). Also updated the changesets (https://github.com/iTwin/iTwinUI/pull/1888/commits/30cceace7ec7c304ad257f87eb640c83e04fd353).